### DISCUSSION TITLE
Keep filters when clicking to another page of the website

### DISCUSSION THREAD

**Discussion by EnterOrion** [vincjo/datatables#94] [Q&A]
Hi,
In my table in my application, each row acts as a link to go to a detailed page about that record. When I click back, the filters on the table are gone. Is there an easy way to keep the filters so that when the user goes back they don't have to retype the filters all over again?
Any help would be appreciated. 

Thank you! And thank you for creating such an awesome library!

**Comment by sergeim19** [comment]
Hi @EnterOrion, did you ever get to resolving this? I am facing the same issue. I was thinking of saving the filters when the user clicks away and then re-applying them when they go back but I am not sure if there is a better method.

**Reply by EnterOrion** [reply]
Unfortunately, no. Your proposed solution is what I ended up doing myself. To help speed it up a little more, I made sure to check if the filter is empty or not before applying it.

**Reply by sergeim19** [reply]
Ok, thanks!

**Comment by vincjo** [comment]
You could use an instance of TableHandler a little higher in your tree.
For example, add in any "index.ts" file:
```ts
import { TableHandler } from '@vincjo/datatables'
export const table = new TableHandler([])
```

In your component, use `table.setRows()` instead of `new TableHandler()`:
```ts
import { table } from '../'

let { data } = $props()

table.setRows(data)
```
This way you can keep user settings after a page change
